packages/common-scripts/tests/p2pkh.test.ts

```diff
@@ -0,0 +1,214 @@
+import test from "ava";
+import { Cell, core, Script, utils } from "@ckb-lumos/base";
+import { TransactionSkeleton } from "@ckb-lumos/helpers";
+import { Reader, normalizers } from "@ckb-lumos/toolkit";
+import * as config from "@ckb-lumos/config-manager";
+import { default as createKeccak } from "keccak";
+import { createP2PKHMessageGroup } from "../src/p2pkh";
+import p2pkhJson from "./p2pkh.json";
+
+const CONFIG = config.createConfig({
+  PREFIX: "ckt",
+  SCRIPTS: {
+    ...config.predefined.AGGRON4.SCRIPTS,
+    OMNI_LOCK: p2pkhJson.SCRIPTS.OMNI_LOCK as config.ScriptConfig,
+    PW_LOCK: p2pkhJson.SCRIPTS.PW_LOCK as config.ScriptConfig,
+  },
+});
+
+test("omni lock [g1]", (t) => {
+  let tx = TransactionSkeleton({});
+
+  const inputCell: Cell = p2pkhJson["OMNI_LOCK_[G1]"].INPUTS as Cell;
+  const outputCell: Cell[] = p2pkhJson["OMNI_LOCK_[G1]"].OUTPUTS as Cell[];
+  tx = tx.update("inputs", (inputs) => inputs.push(inputCell));
+  tx = tx.update("outputs", (outputs) => outputs.push(...outputCell));
+
+  tx = tx.update("cellDeps", (cellDeps) =>
+    cellDeps.push(
+      {
+        out_point: {
+          tx_hash: CONFIG.SCRIPTS.OMNI_LOCK.TX_HASH,
+          index: CONFIG.SCRIPTS.OMNI_LOCK.INDEX,
+        },
+        dep_type: CONFIG.SCRIPTS.OMNI_LOCK.DEP_TYPE,
+      },
+      {
+        out_point: {
+          tx_hash: CONFIG.SCRIPTS.SECP256K1_BLAKE160.TX_HASH,
+          index: CONFIG.SCRIPTS.SECP256K1_BLAKE160.INDEX,
+        },
+        dep_type: CONFIG.SCRIPTS.SECP256K1_BLAKE160.DEP_TYPE,
+      }
+    )
+  );
+
+  const SECP_SIGNATURE_PLACEHOLDER = new Reader("0x" + "00".repeat(85));
+  const tmpWitnessArgs = { lock: SECP_SIGNATURE_PLACEHOLDER };
+  const tmpWitness = new Reader(
+    core.SerializeWitnessArgs(normalizers.NormalizeWitnessArgs(tmpWitnessArgs))
+  ).serializeJson();
+  tx = tx.update("witnesses", (witnesses) =>
+    witnesses.push(tmpWitness, tmpWitness)
+  );
+
+  const hasher = new utils.CKBHasher();
+  const signLock = p2pkhJson["OMNI_LOCK_[G1]"].SIGN_LOCK as Script;
```

Comment:
maybe we should extract this section as a function, and put the `tmpWitness` into the fixture file

```ts
function txSkeletonFromJson(tx: TransactionSkeletonType): object;
function txSkeletonToJson(object: object): TransactionSkeletonType;
```

```ts



const skeleton = txSkeletonFromJson(p2pkhFixtures['some case'].tx);
const signLock = {...}
createP2PKHMessageGroup(skeleton, [signLock], p2pkhFixtures['some case'].expectedMessage)
```